pkg/bucketeer/user.go

```diff
@@ -1,20 +1,43 @@
 package bucketeer
 
-import "github.com/ca-dp/bucketeer-go-server-sdk/proto/user"
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ca-dp/bucketeer-go-server-sdk/pkg/bucketeer/internal/uuid"
+	protouser "github.com/ca-dp/bucketeer-go-server-sdk/proto/user"
+)
 
 // User is the Bucketeer user.
 //
 // User contains a mandatory user id and optional attributes (Data) for the user targeting and the analysis.
 type User struct {
-	*user.User
+	*protouser.User
 }
 
-// NewUser is the constructor for User.
+// NewUser creates a new User.
 //
 // id is mandatory and attributes is optional.
-func NewUser(id string, attributes map[string]string) *User {
-	return &User{User: &user.User{
+func NewUser(id string, attributes map[string]string) (*User, error) {
+	if id == "" {
+		return nil, errors.New("bucketeer: id is empty")
+	}
+	return &User{User: &protouser.User{
 		Id:   id,
 		Data: attributes,
-	}}
+	}}, nil
+}
+
+// NewRandomUser creates a new User with a random id.
+//
+// attributes is optional.
+func NewRandomUser(attributes map[string]string) (*User, error) {
```

Comment:
Where is this function going to be used?

Reply:
I thought some customers (for example, they don't manage user id) wanted to use this function, but it may be unnecessary.
I'm going to remove it.